tests/Admin/AdminTest.php

```diff
@@ -1794,6 +1798,88 @@ public function testGetFilterFieldDescription(): void
         $this->assertSame($bazFieldDescription, $modelAdmin->getFilterFieldDescription('baz'));
     }
 
+    public function testGetDatagrid(): void
+    {
+        $modelManager = $this->createMock(ModelManagerInterface::class);
+        $datagridBuilder = $this->createMock(DatagridBuilderInterface::class);
+        $pager = $this->createMock(PagerInterface::class);
+        $datagrid = $this->createMock(DatagridInterface::class);
+
+        $admin = new CleanAdmin('test code', 'Application\Entity\CleanEntity', 'SonataFooBundle:CleanAdmin');
+        $admin->setModelManager($modelManager);
+        $admin->setDatagridBuilder($datagridBuilder);
+
+        $datagrid->expects($this->once())
+            ->method('getPager')
+            ->willReturn($pager)
+        ;
+
+        $datagridBuilder->expects($this->once())
+            ->method('getBaseDatagrid')
+            ->with($this->identicalTo($admin), [
+                '_sort_by'    => null,
+                '_sort_order' => null,
+            ])
+            ->willReturn($datagrid)
+        ;
+
+        $result=$admin->getDatagrid();
+        self::assertNotNull($result);
+    }
+
+    public function testGetDatagridWithCustomQuery(): void
+    {
+        $modelManager = $this->getMockForAbstractClass(ModelManagerInterface::class);
+        $listBuilder = $this->getMockForAbstractClass(ListBuilderInterface::class);
+        $pager = $this->createMock(PagerInterface::class);
+        $datagrid = $this->createMock(DatagridInterface::class);
+        $datagridBuilder = $this->createMock(DatagridBuilderInterface::class);
+
+        $modelManager
+            ->expects(self::once())
+            ->method('createQuery')
+            ->willReturn(new CustomQuery());
```

Comment:
Why not using a mock of `ProxyQueryInterface` ?

Reply:
Because of an error:
```php
Error : Call to a member function setSortBy() on null
tests/Fixtures/Admin/CustomQueryAdmin.php:21
src/Admin/AbstractAdmin.php:3139
src/Admin/AbstractAdmin.php:813
src/Admin/AbstractAdmin.php:1355
tests/Admin/AdminTest.php:1878
```

Reply:
You just need to mock the function you need 
```
$query = $this->createMock(ProxyQueryInterface::class);
$query->expects($this->once())
            ->method('setSortOrder')
            ->with(...)
            ->willReturn($query);
$query->expects($this->once())
            ->method('setSortBy')
            ->with(...)
            ->willReturn($query);
```

Reply:
Done (26517feb commit)